### PR TITLE
add an extra check for paint member of style layer

### PR DESCRIPTION
**Pull request by mannnick24**
We are getting a browser dependent, intermittent error from fill_style_layer.js:45
the query radius function expects paint memeber to be truthy and its not
so we have a TypeError: this.paint is undefined

I've been trying for a couple of days to reproduce this in a fiddle and I can't (yet)

### CODE REVIEW

File: src/source/tile.js [hunk 1]
```diff
@@ -404,7 +404,8 @@ class Tile {
             if (!sourceLayer || !sourceLayerStates || Object.keys(sourceLayerStates).length === 0) continue;
 
             bucket.update(sourceLayerStates, sourceLayer, this.imageAtlas && this.imageAtlas.patternPositions || {});
-            if (painter && painter.style) {
+
+            if (painter && painter.style && painter.style.getLayer(id).paint) {
                 this.queryPadding = Math.max(this.queryPadding, painter.style.getLayer(id).queryRadius(bucket));
```
Comment (mourner) [inline src/source/tile.js]:
```suggestion
            const layer = painter.style.getLayer(id);
            if (painter && painter.style && layer.paint) {
                this.queryPadding = Math.max(this.queryPadding, layer.queryRadius(bucket));
```

**Comment by fminuti** [comment]
Sorry, but what's the meaning of first use "painter.style" and THEN check it (inside the IF)?

**Comment by mannnick24** [comment]
@fminuti quite right... checking painter && painter.style again is not needed, I'd be hesitant to change someone else's codebase though :-)
I'd probably get the layer = painter.style.getLayer(id) at the top and check its truthy instead of the has

**Comment by mourner** [comment]
@fminuti good catch! My bad. Submitted #9472 to follow up.

**Pull request merged**